### PR TITLE
Clean up C++ issues exposed by static analysis

### PR DESCRIPTION
**Pull request by cmcantalupo**
- Relates to #2259

### CODE REVIEW

**Review by bgeltz — DISMISSED** [review]
LGTM.  It's probably a good idea to at least run test_monitor.py before merging since there were changes to MSRIO.

File: service/src/Exception.cpp [hunk 1]
```diff
@@ -103,6 +103,12 @@ namespace geopm
 
     }
 
+    Exception &Exception::operator=(const Exception &other)
+    {
+        m_err = other.m_err;
+        return *this;
+    }
```
Comment (avilcheslopez) [inline service/src/Exception.cpp:111]: Should this call the base class' assignment operator as well?

**Review by avilcheslopez — COMMENTED** [review]
Just a minor observation

**Review by avilcheslopez — APPROVED** [review]
Looks good to me!

File: service/src/Exception.cpp (hunk 1 above)
Comment (cmcantalupo) [reply service/src/Exception.cpp:111]: Thanks!

**Pull request merged**